### PR TITLE
feat: add next banner

### PR DESCRIPTION
**Pull request by marthayaputra**

### CODE REVIEW

File: src/@next/Icon/icons/icons.ts [hunk 1]
```diff
@@ -67,6 +67,13 @@ export const iconNames = [
   'ri-send-plane-line',
   'ri-settings-fill',
   'ri-settings-line',
+  'ri-checkbox-fill',
+  'ri-checkbox-line',
+  'ri-checkbox-indeterminate-fill',
+  'ri-checkbox-indeterminate-line',
+  'ri-search',
+  'ri-loader',
```
Comment (ninariccimarie) [inline src/@next/Icon/icons/icons.ts]: duplicates?

File: src/@next/Icon/icons/icons.ts [hunk 2]
```diff
@@ -67,6 +67,13 @@ export const iconNames = [
...
+  'ri-checkbox-circle-fill',
```
Comment (ninariccimarie) [inline src/@next/Icon/icons/icons.ts]: can you move it above to be in alphabetical order?

File: src/@next/Icon/icons/icons.ts [hunk 3]
```diff
@@ -139,5 +146,6 @@ export const iconsMappingComponent: { [name in IconNames]: SVGComponent } = {
   ['ri-send-plane-line']: Icons.RiSendPlaneLine,
   ['ri-settings-fill']: Icons.RiSettingsFill,
   ['ri-settings-line']: Icons.RiSettingsLine,
+  ['ri-checkbox-circle-fill']: Icons.RiCheckboxCircleFill,
```
Comment (ninariccimarie) [inline src/@next/Icon/icons/icons.ts]: can you also move this above to be in alphabetical order?

File: src/@next/Banner/index.tsx [hunk 4]
```diff
@@ -0,0 +1 @@
+export { Banner, BannerProps } from './Banner';
```
Comment (ninariccimarie) [inline src/@next/Banner/index.tsx:1]: any reason we exported BannerProps here?

File: src/@next/Banner/Banner.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  const iconByStatus = iconNameStatusMap[status];
+  if (!iconByStatus) {
+    console.warn(`Status ${status} is not valid, default style will be used`);
+  }
+  const iconNameValue = iconName
+    ? iconName
+    : iconByStatus || iconNameStatusMap['info'];
+
+  return (
+    <div>
```
Comment (ninariccimarie) [inline src/@next/Banner/Banner.tsx]: do we still need this div?

File: src/@next/Banner/Banner.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  iconName?: IconNames;
+  status?: 'success' | 'info' | 'warning' | 'critical';
+  children?: React.ReactNode;
+  action?: React.ReactNode;
+  secondaryAction?: React.ReactNode;
+  dismissable?: boolean;
+  onDismiss?: () => void;
+};
+
+const iconNameStatusMap = {
```
Comment (ninariccimarie) [inline src/@next/Banner/Banner.tsx]: maybe we can type this so we don't need to cast the type of `iconNameValue`. we can use `Partial<IconNames>` maybe?

File: src/@next/Banner/BannerStyle.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
+import styled from 'styled-components';
+import * as Breakpoints from '../utilities/breakpoints';
+import { borderRadius8 } from '../utilities/borderRadius';
+import { Blue, Green, Neutral, Orange, Red } from '../utilities/colors';
+
+export const StyledCloseIconWrapper = styled.div`
```
Comment (ninariccimarie) [inline src/@next/Banner/BannerStyle.ts:6]: why don't we use button instead of div?

File: src/@next/Banner/BannerStyle.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  cursor: pointer;
+  svg {
+    right: 0;
+    top: 0 !important;
```
Comment (ninariccimarie) [inline src/@next/Banner/BannerStyle.ts]: would it be worth writing it this way instead? https://styled-components.com/docs/faqs#how-can-i-override-styles-with-higher-specificity

File: src/@next/Icon/icons/icons.ts (hunk 1 above)
Comment (marthayaputra) [reply src/@next/Icon/icons/icons.ts]: Oh yes, thanks for that

File: src/@next/Icon/icons/icons.ts (hunk 2 above)
Comment (marthayaputra) [reply src/@next/Icon/icons/icons.ts]: moved

File: src/@next/Icon/icons/icons.ts (hunk 3 above)
Comment (marthayaputra) [reply src/@next/Icon/icons/icons.ts]: moved

File: src/@next/Banner/index.tsx (hunk 4 above)
Comment (marthayaputra) [reply src/@next/Banner/index.tsx:1]: Yeah, I think it's helpful if we export the props with components actually, probably need to do that for others later. I realized I often need third party component to type my abstracted layer of component above them.

File: src/@next/Banner/Banner.tsx (hunk 6 above)
Comment (marthayaputra) [reply src/@next/Banner/Banner.tsx]: Yeah good call!

File: src/@next/Banner/BannerStyle.ts (hunk 7 above)
Comment (marthayaputra) [reply src/@next/Banner/BannerStyle.ts:6]: Shorter styling, will need to remove some natural button attributes.  Do you see any risk with this?

File: src/@next/Banner/BannerStyle.ts (hunk 8 above)
Comment (marthayaputra) [reply src/@next/Banner/BannerStyle.ts]: Oh, didn't know about this, let me try. Thanks

File: src/@next/Banner/Banner.tsx (hunk 5 above)
Comment (marthayaputra) [reply src/@next/Banner/Banner.tsx]: nope, thanks!

File: src/@next/Banner/index.tsx (hunk 4 above)
Comment (ninariccimarie) [reply src/@next/Banner/index.tsx:1]: yeah, I experienced the same. then we probably need to export it in [@next/index ](https://github.com/glints-dev/glints-aries/pull/718/files#diff-5b6ecadc6bce07b9e2cf58256d9853ac20fa4c04f755c55979c70f0620032ec7R25) instead?

File: src/@next/Banner/BannerStyle.ts (hunk 7 above)
Comment (ninariccimarie) [reply src/@next/Banner/BannerStyle.ts:6]: i did read that buttons are harder to style vs divs. common risk would be accessibility if we use divs

Comment (marthayaputra) [reply src/@next/Banner/BannerStyle.ts:6]: Oh, I added role "button" there to address that

File: src/@next/Banner/index.tsx (hunk 4 above)
Comment (marthayaputra) [reply src/@next/Banner/index.tsx:1]: Oh crap, I tot I did, thanks!

**Review by ninariccimarie — APPROVED** [review]
LGTM 🚀

**Pull request merged**